src/chttpd/src/chttpd_httpd_handlers.erl

```diff
@@ -48,27 +48,439 @@ design_handler(<<"_info">>)    -> fun chttpd_db:handle_design_info_req/3;
+    }};
+
+handler_info('DELETE', [Db, <<"_local">>, Name], _) ->
+    {'db.local.doc.delete', #{'db.name' => Db, 'local.id' => Name}};
+
+handler_info(_, [Db, <<"_local">>, Name | _Path], _) ->
+    {'db.local.doc.invalid_attachment_req', #{
+        'db.name' => Db,
+        'local.id' => Name
+    }};
+
+handler_info(M, [Db, <<"_local_docs">>], _) when M == 'GET'; M == 'POST' ->
+    {'db.local_docs.read', #{'db.name' => Db}};
+
+handler_info('POST', [Db, <<"_local_docs">>, <<"queries">>], _) ->
+    {'db.local_docs.read', #{'db.name' => Db, multi => true}};
+
+handler_info('POST', [Db, <<"_missing_revs">>], _) ->
+    {'db.docs.missing_revs.execute', #{'db.name' => Db}};
+
+handler_info('GET', [Db, <<"_partition">>, Partition], _) ->
+    {'db.partition.info.read', #{'db.name' => Db, partition => Partition}};
+
+handler_info(_, [Db, <<"_partition">>, Partition | Rest], Req) ->
+    NewPath = case Rest of
+        [<<"_all_docs">> | _] ->
+            [Db | Rest];
+        [<<"_index">> | _] ->
+            [Db | Rest];
+        [<<"_find">> | _] ->
+            [Db | Rest];
+        [<<"_explain">> | _] ->
+            [Db | Rest];
+        [<<"_design">>, _Name, <<"_", _/binary>> | _] ->
+            [Db | Rest];
+        _ ->
+            no_match
+    end,
+    if NewPath == no_match -> no_match; true ->
+        {OpName, Tags} = chttpd_handlers:handler_info(Req#httpd{
+            path_parts = NewPath
+        }),
+        NewOpName = case atom_to_list(OpName) of
+            "db." ++ Name -> list_to_atom("db.partition." ++ Name);
+            Else -> list_to_atom(Else ++ ".partition")
+        end,
+        {NewOpName, maps:put(partition, Partition, Tags)}
+    end;
+
+handler_info('POST', [Db, <<"_purge">>], _) ->
+    {'db.docs.purge', #{'db.name' => Db}};
+
+handler_info('GET', [Db, <<"_purged_infos_limit">>], _) ->
+    {'db.purged_infos_limit.read', #{'db.name' => Db}};
+
+handler_info('PUT', [Db, <<"_purged_infos_limit">>], _) ->
+    {'db.purged_infos_limit.write', #{'db.name' => Db}};
+
+handler_info('POST', [Db, <<"_revs_diff">>], _) ->
+    {'db.docs.revs_diff.execute', #{'db.name' => Db}};
+
+handler_info('GET', [Db, <<"_revs_limit">>], _) ->
+    {'db.revs_limit.read', #{'db.name' => Db}};
+
+handler_info('PUT', [Db, <<"_revs_limit">>], _) ->
+    {'db.revs_limit.write', #{'db.name' => Db}};
+
+handler_info('GET', [Db, <<"_security">>], _) ->
+    {'db.security.read', #{'db.name' => Db}};
+
+handler_info('PUT', [Db, <<"_security">>], _) ->
+    {'db.security.write', #{'db.name' => Db}};
+
+handler_info(_, [Db, <<"_view_cleanup">>], _) ->
+    {'views.cleanup.execute', #{'db.name' => Db}};
+
+handler_info(_, [_Db, <<"_", _/binary>> | _], _) ->
+    % Bail here for other possible db_handleres
+    no_match;
+
+handler_info('GET', [Db, DocId], _) ->
+    {'db.doc.read', #{'db.name' => Db, 'doc.id' => DocId}};
+
+handler_info('POST', [Db, DocId], _) ->
+    {'db.doc.write', #{'db.name' => Db, 'design.id' => DocId}};
```

Comment:
Like in the ```_design``` doc case, does this need to be ```handler_info('POST', [Db],```

Reply:
Were you thinking about this line?

https://github.com/apache/couchdb/pull/2323/files#diff-366177ccb23bdbd2f8321c8dcc6811f7R187-R188

The rough progression of pattern matching clauses is to be in both alphabetical and length sorted order. I.e., it starts with [], and then does all of the `_action` bits then `[Db]`, then `[Db, _action]` etc etc. Which on the one hand means its easy to find what we're looking for. Though on the other hand it also means that related handlers may be far apart.

Given that its such a huge pattern match I'd keep the current sorting, but maybe adding an description of the ordering would help?

Reply:
I just meant that not sure about POSTing to /db/docid is not a valid API endpoint 

http://docs.couchdb.org/en/stable/api/document/common.html#db-doc

Reply:
Its not a common one

https://github.com/apache/couchdb/blob/prototype/fdb-layer/src/chttpd/src/chttpd_db.erl#L1071-L1124